cernopendata_client/cli.py

```diff
@@ -181,11 +188,21 @@ def get_file_locations(server, recid, doi, title, protocol, expand):
 )
 @click.option(
     "--filter-range",
-    "range",
+    "ranges",
+    multiple=True,
     type=click.STRING,
     help="Download files from a specified list range (i-j)",
 )
-def download_files(server, recid, doi, title, protocol, expand, name, regexp, range):
+@click.option(
+    "--verify",
+    "verify",
+    is_flag=True,
+    default=False,
```

Comment:
I was wondering whether we shouldn't trigger verification by default, but then I noticed that it does not play nicely with filters:

```console
$ cernopendata-client download-files --recid 5500 --filter-name BuildFile.xml
==> Downloading file 1 of 1
  -> File: ./5500/BuildFile.xml
  -> Progress: 0/0 kiB (100%)
==> Success!
$ $ cernopendata-client download-files --recid 5500 --filter-name BuildFile.xml --verify

==> Downloading file 1 of 1
  -> File: ./5500/BuildFile.xml
  -> Progress: 0/0 kiB (100%)
==> Verifying downloaded files for record 5500.
  -> expected 1, found 11.
ERROR: File count does not match.
```

I would expect the following: 
- first filter which files user wants to download
- then for each file to be downloaded, do:
   - download it
   - check its size and sha1
- success

This means that we don't have to verify total number of files... only count and checksum for each individual downloaded file.

(And, since the verification of fast, perhaps we can switch it on by default?)

Reply:
Created a new issue #62 for improving it.